src/mono/mono/mini/mini-llvm.c

```diff
@@ -8439,8 +8439,9 @@ MONO_RESTORE_WARNING
 
 		case OP_SSE_LOADU: {
 			LLVMValueRef dst_ptr = convert (ctx, lhs, LLVMPointerType (primitive_type_to_llvm_type (inst_c1_type (ins)), 0));
-			LLVMValueRef dst_vec = LLVMBuildBitCast (builder, dst_ptr, LLVMPointerType (type_to_sse_type (ins->inst_c1), 0), "");
-			values [ins->dreg] = mono_llvm_build_aligned_load (builder, dst_vec, "", FALSE, ins->inst_c0); // inst_c0 is alignment
+			LLVMTypeRef etype = type_to_sse_type (ins->inst_c1);
+			LLVMValueRef dst_vec = LLVMBuildBitCast (builder, dst_ptr, LLVMPointerType (etype, 0), "");
```

Comment:
one of the motivations for opaque pointer types in the LLVM docs is that it allows them to get rid of a lot of pointer-to-pointer bitcasts that are just changing the element type.  

I'm guessing they probably have a pass that will remove these pretty early on.

Do you think we should try to avoid emitting them at all?

Reply:
We are still using the non-opaque mode since the opaque mode is experimental afaik in llvm 11.x. So we can remove those after switching to llvm 14.x.